### PR TITLE
fix: rename files asynchronously 

### PR DESCRIPTION
**Pull request by raredeveloper**

### CODE REVIEW

File: app/src/main/java/com/itsaky/androidide/actions/filetree/RenameAction.kt [hunk 1]
```diff
@@ -66,27 +68,34 @@ class RenameAction(context: Context, override val order: Int) :
...
-      flashMessage(
-        if (renamed) com.itsaky.androidide.resources.R.string.renamed
-        else com.itsaky.androidide.resources.R.string.rename_failed,
-        if (renamed) FlashType.SUCCESS else FlashType.ERROR
-      )
-      if (!renamed) {
-        return@setPositiveButton
-      }
+      val progressDialog =
+          ProgressDialog.show(context, null, context.getString(com.itsaky.androidide.resources.R.string.please_wait), true, false)
```
Comment (MrIkso) [inline app/src/main/java/com/itsaky/androidide/actions/filetree/RenameAction.kt:72]: ProgressDialog is deprecated

Comment (raredeveloper) [reply app/src/main/java/com/itsaky/androidide/actions/filetree/RenameAction.kt:72]: The same is used in "DeleteAction".

File: app/src/main/java/com/itsaky/androidide/actions/filetree/RenameAction.kt [hunk 2]
```diff
@@ -66,27 +67,30 @@ class RenameAction(context: Context, override val order: Int) :
...
-      val renamed = name.length in 1..40 && FileUtils.rename(file, name)
-      flashMessage(
-        if (renamed) com.itsaky.androidide.resources.R.string.renamed
-        else com.itsaky.androidide.resources.R.string.rename_failed,
-        if (renamed) FlashType.SUCCESS else FlashType.ERROR
-      )
-      if (!renamed) {
-        return@setPositiveButton
-      }
+      executeAsync({ name.length in 1..40 && FileUtils.rename(file, name) }) {
```
[the remark this answers is not in the document]
Comment (raredeveloper) [reply app/src/main/java/com/itsaky/androidide/actions/filetree/RenameAction.kt:70]: Okay

[the remark this answers is not in the document]
Comment (raredeveloper) [reply app/src/main/java/com/itsaky/androidide/actions/filetree/RenameAction.kt]: There is a problem: if the user collapses the parent node and expands it again while renaming the renamed file node will not be updated

File: app/src/main/java/com/itsaky/androidide/actions/filetree/RenameAction.kt [hunk 3]
```diff
@@ -65,29 +68,37 @@ class RenameAction(context: Context, override val order: Int) :
...
-        val parent = lastHeld.parent
-        parent.deleteChild(lastHeld)
-        val node = TreeNode(File(file.parentFile, name))
-        node.viewHolder = FileTreeViewHolder(context)
-        parent.addChild(node)
-        requestExpandNode(parent)
-      } else {
-        requestFileListing()
-      }
+              notifyFileRenamed(file, name, context)
```
Comment (itsaky) [inline app/src/main/java/com/itsaky/androidide/actions/filetree/RenameAction.kt:88]: You can safely move `notifyFileRenamed` outside of the `Main` context. For `notifyFileRenamed`, EventBus will dispatch the event on the event receiver's preferred thread.

**Review by itsaky — CHANGES_REQUESTED** [review]

**Review by itsaky — APPROVED** [review]

**Comment by itsaky** [comment]
Thank you for your contributions.

**Pull request merged**